### DISCUSSION TITLE
Invalid restore input. No target frameworks specified.

### DISCUSSION THREAD

**Discussion by montella1507** [unoplatform/uno#8879] [Q&A]
### Current behavior

I have done all regarding the documentation here:
https://platform.uno/docs/articles/get-started-rider.html?tabs=macos

When i run dotnet restore i got error:
```
UNO/MyFirstUnoApp/MyFirstUnoApp.Droid/MyFirstUnoApp.Droid.csproj: Invalid restore input. No target frameworks specified. Input files: MyFirstUnoApp/MyFirstUnoApp.Droid/MyFirstUnoApp.Droid.csproj.
UNO/MyFirstUnoApp/MyFirstUnoApp.iOS/MyFirstUnoApp.iOS.csproj: Invalid restore input. No target frameworks specified. UNO/MyFirstUnoApp/MyFirstUnoApp.iOS/MyFirstUnoApp.iOS.csproj.
UNO/MyFirstUnoApp/MyFirstUnoApp.macOS/MyFirstUnoApp.macOS.csproj: Invalid restore input. No target frameworks specified. Input files: UNO/MyFirstUnoApp/MyFirstUnoApp.macOS/MyFirstUnoApp.macOS.csproj.
UNO/MyFirstUnoApp/MyFirstUnoApp.UWP/MyFirstUnoApp.Uwp.csproj: Invalid restore input. No target frameworks specified. Input files: UNO/MyFirstUnoApp/MyFirstUnoApp.UWP/MyFirstUnoApp.Uwp.csproj.
```

### Expected behavior

It should be able to restore packages in generated project by 
dotnet new unoapp -o MyApp

### Environment

Other

### Affected platforms

macOS

### IDE

Rider macOS

### Anything else we need to know?

M1 Macbook PRO, macOS 12.4

dotnet --list-sdks
6.0.100 [/usr/local/share/dotnet/sdk]
6.0.300 [/usr/local/share/dotnet/sdk]

dotnet --list-runtimes
Microsoft.AspNetCore.App 6.0.0 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
Microsoft.AspNetCore.App 6.0.5 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
Microsoft.NETCore.App 6.0.0 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
Microsoft.NETCore.App 6.0.5 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]

dotnet restore fails both in console / rider

**Comment by montella1507** [comment]
Here is the generated project
[MyFirstUnoApp.zip](https://github.com/unoplatform/uno/files/8770781/MyFirstUnoApp.zip)

**Reply by Inaganti97** [reply]
upgrade-assistant upgrade /Users/yalamandaraoinaganti/Downloads/MyFirstUnoApp/MyFirstUnoApp.iOS/MyFirstUnoApp.iOS.csproj

What is your preferred target framework?                                        
                                                                                
> There are no target frameworks found to which your project could be upgraded. 

Do you know how to resolve this one ? I'm also getting same issue in my xamarin native projects

**Comment by jeromelaban** [comment]
This would be an issue specific to rider. Projects created by the templates are standard Xamarin.iOS/Android projects. Make sure that standard xamarin projects are working in your context first.

**Comment by montella1507** [comment]
No it isnt. I got this error first time when i:
1) GENERATED project in terminal - it runs dotnet restore automatically after the project is generated
2) Second time when i write "dotnet restore" second time to the terminal.

AFTER that i HOPED rider will solve this issue magicaly.. but i got the same Error.

**Comment by montella1507** [comment]
```
marianbencat@Marians-MacBook-Pro pepa % dotnet new unoapp -o MyApp
The template "Multi-Platform App" was created successfully.

Processing post-creation actions...
Running 'dotnet restore' on /Users/marianbencat/RiderProjects/pepa/MyApp/MyApp.Shared/MyApp.Shared.shproj...
Restore succeeded.
Running 'dotnet restore' on /Users/marianbencat/RiderProjects/pepa/MyApp/MyApp.Droid/MyApp.Droid.csproj...
Restore succeeded.
Running 'dotnet restore' on /Users/marianbencat/RiderProjects/pepa/MyApp/MyApp.iOS/MyApp.iOS.csproj...
Restore succeeded.
Running 'dotnet restore' on /Users/marianbencat/RiderProjects/pepa/MyApp/MyApp.UWP/MyApp.Uwp.csproj...
Restore succeeded.
Running 'dotnet restore' on /Users/marianbencat/RiderProjects/pepa/MyApp/MyApp.macOS/MyApp.macOS.csproj...
Restore succeeded.
Running 'dotnet restore' on /Users/marianbencat/RiderProjects/pepa/MyApp/MyApp.Skia.Gtk/MyApp.Skia.Gtk.csproj...
  Determining projects to restore...
  Restored /Users/marianbencat/RiderProjects/pepa/MyApp/MyApp.Skia.Gtk/MyApp.Skia.Gtk.csproj (in 338 ms).
Restore succeeded.
Running 'dotnet restore' on /Users/marianbencat/RiderProjects/pepa/MyApp/MyApp.Skia.WPF/MyApp.Skia.Wpf.csproj...
  Determining projects to restore...
/Users/marianbencat/RiderProjects/pepa/MyApp/MyApp.Skia.WPF/MyApp.Skia.Wpf.csproj : warning NU1701: Package 'SkiaSharp.Views.WPF 2.80.3' was restored using '.NETFramework,Version=v4.6.1, .NETFramework,Version=v4.6.2, .NETFramework,Version=v4.7, .NETFramework,Version=v4.7.1, .NETFramework,Version=v4.7.2, .NETFramework,Version=v4.8' instead of the project target framework '.NETStandard,Version=v2.0'. This package may not be fully compatible with your project.
  Restored /Users/marianbencat/RiderProjects/pepa/MyApp/MyApp.Skia.WPF/MyApp.Skia.Wpf.csproj (in 128 ms).
Restore succeeded.
Running 'dotnet restore' on /Users/marianbencat/RiderProjects/pepa/MyApp/MyApp.Skia.WPF.Host/MyApp.Skia.Wpf.Host.csproj...
  Determining projects to restore...
/usr/local/share/dotnet/sdk/6.0.300/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.Sdk.FrameworkReferenceResolution.targets(90,5): error NETSDK1100: Windows is required to build Windows desktop applications. [/Users/marianbencat/RiderProjects/pepa/MyApp/MyApp.Skia.WPF.Host/MyApp.Skia.Wpf.Host.csproj]
Restore failed.
Running 'dotnet restore' on /Users/marianbencat/RiderProjects/pepa/MyApp/MyApp.Skia.Tizen/MyApp.Skia.Tizen.csproj...
  Determining projects to restore...
  Restored /Users/marianbencat/RiderProjects/pepa/MyApp/MyApp.Skia.Tizen/MyApp.Skia.Tizen.csproj (in 75 ms).
Restore succeeded.
Running 'dotnet restore' on /Users/marianbencat/RiderProjects/pepa/MyApp/MyApp.Skia.Linux.FrameBuffer/MyApp.Skia.Linux.FrameBuffer.csproj...
  Determining projects to restore...
  Restored /Users/marianbencat/RiderProjects/pepa/MyApp/MyApp.Skia.Linux.FrameBuffer/MyApp.Skia.Linux.FrameBuffer.csproj (in 60 ms).
Restore succeeded.
Post action failed.
Manual instructions: Run 'dotnet restore'
```

**Comment by jeromelaban** [comment]
Thanks. Does a vanilla Xamarin.iOS/Android app build in rider? Once created from the templates in visual studio for mac, for instance.

**Comment by montella1507** [comment]
Seems there are 2 problems:
- you  are right about rider, there is "no target frameworks" error message only in rider, when i installed additional plugins (mono etc) the error is gone.
- second error is  with "dotnet restore" as well, but the error is with Skia.WPF.Host project 

`error NETSDK1100: Windows is required to build Windows desktop applications.`

i think this SHOULD NOT be threated as an error, just warning? After i have removed the project from solution i am able to build that. However i am still not able to run it:

"Error running 'MyFirstUnoApp.macOS': Failed to launch application: file 'com.jetbrains.rider.run.configurations.xamarin.mac.XamarinMacExecutable@6ea114ec' doesn't exist"

but this may be problem with Xcode installation or something similard.

So we may close this issue - it may be good to add note to the "Getting started" page that user has to remove unsupported projects from generated code, else the dotnet restore will fail?

**Comment by jeromelaban** [comment]
The error message you're seeing is something the .NET SDK is giving, it's not something we have control over. You can exclude the   wpf head if you want to, by using the [dotnet new switches](https://platform.uno/docs/articles/get-started-dotnet-new.html?tabs=windows#uno-platform-blank-application).

You can use `dotnet new unoapp -h` to see more details. You can also try with the `dotnet new unoapp-winui-net6` template, but I don't know if rider supports it yet.
